core/fslock/fslock.cpp

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "fslock/fslock.hpp"
+#include <boost/filesystem/fstream.hpp>
+#include <boost/filesystem/operations.hpp>
+#include "fslock/fslock_error.hpp"
+
+namespace fc::fslock {
+  outcome::result<boost::interprocess::file_lock> lock(
+      const std::string &lock_file_path) {
+    boost::filesystem::ofstream ofs(lock_file_path.c_str(), std::ios::app);
+    ofs.close();
+    boost::interprocess::file_lock lock_file;
+    try {
+      lock_file = boost::interprocess::file_lock(lock_file_path.c_str());
+    } catch (const boost::interprocess::interprocess_exception &e) {
+      return FSLockError::NO_RESOURCES;
+    }
```

Comment:
} catch (...) { // so you will catch any remaining exception
  return FSLockError::UNKNOWN_ERROR; // or smth
}

Reply:
Fixed